src/main/kotlin/org/opensearch/observability/index/ObservabilityIndex.kt

```diff
@@ -96,32 +96,41 @@ internal object ObservabilityIndex : LifecycleListener() {
      */
     @Suppress("TooGenericExceptionCaught")
     private fun createIndex() {
-        if (!isIndexExists(INDEX_NAME)) {
-            val classLoader = ObservabilityIndex::class.java.classLoader
-            val indexMappingSource = classLoader.getResource(OBSERVABILITY_MAPPING_FILE_NAME)?.readText()!!
-            val indexSettingsSource = classLoader.getResource(OBSERVABILITY_SETTINGS_FILE_NAME)?.readText()!!
-            val request = CreateIndexRequest(INDEX_NAME)
-                .mapping(indexMappingSource, XContentType.YAML)
-                .settings(indexSettingsSource, XContentType.YAML)
-            try {
-                val actionFuture = client.admin().indices().create(request)
-                val response = actionFuture.actionGet(PluginSettings.operationTimeoutMs)
-                if (response.isAcknowledged) {
-                    log.info("$LOG_PREFIX:Index $INDEX_NAME creation Acknowledged")
-                    reindexNotebooks()
-                } else {
-                    error("$LOG_PREFIX:Index $INDEX_NAME creation not Acknowledged")
-                }
-            } catch (exception: ResourceAlreadyExistsException) {
-                log.warn("message: ${exception.message}")
-            } catch (exception: Exception) {
-                if (exception.cause !is ResourceAlreadyExistsException) {
-                    throw exception
+        try {
+            // wait for the cluster here until it will finish managed node election
+            while (clusterService.state().blocks().hasGlobalBlockWithStatus(RestStatus.SERVICE_UNAVAILABLE)) {
+                log.info("Wait for cluster to be available ...")
```

Comment:
Thanks for the fix! 
I had the same question that was raised by Shwetha in the original issue:
```
 The main branch doesn't have same code as well? why is there divergence?
```
Do we know why was this only needed for 2.x and what's the path ahead to converge these branches?

Reply:
@ps48   I'm not sure why this was removed in the main version
not really involved recently with this repository....